makeflow/src/makeflow_archive_query.py

```diff
@@ -0,0 +1,247 @@
+import hashlib
```

Comment:
Call the program `makeflow_archive_query` and put a pound-bang /usr/bin/python at the top;
from a UX perspective, the user shouldn't have to care what language it's written in.